chunks 1-3: Tweak page headers (local change).
chunk 4: Default value for the .Os macro (local change).
chunk 5: Accept any number of arguments for .Dd (committed upstream, too).
chunk 6: Do not use an ugly double dash in the NAME line (local change).

Index: tmac/doc-common-u
--- tmac/doc-common-u.orig
+++ tmac/doc-common-u
@@ -234,12 +234,10 @@
 .
 .\" an alternative, more detailed scheme for naming the manual sections
 .\"
-.ds doc-volume-operating-system BSD
-.nr doc-volume-operating-system-ateol 0
 .ds doc-volume-ds-1 General Commands Manual
 .ds doc-volume-ds-2 System Calls Manual
 .ds doc-volume-ds-3 Library Functions Manual
-.ds doc-volume-ds-4 Kernel Interfaces Manual
+.ds doc-volume-ds-4 Device Drivers Manual
 .ds doc-volume-ds-5 File Formats Manual
 .ds doc-volume-ds-6 Games Manual
 .ds doc-volume-ds-7 Miscellaneous Information Manual
@@ -358,19 +356,10 @@
 .    ds doc-section \$2
 .    ie \B\$2 \{\
 .      if ((\$2 >= 1) & (\$2 <= 9)) \{\
-.        ie \n[doc-volume-operating-system-ateol] \{\
-.          ds doc-volume "\*[doc-volume-ds-\$2] \*[doc-volume-operating-system]
-.          if \A\$3 \{\
-.            if d doc-volume-as-\$3 \
-.              as doc-volume "/\*[doc-volume-as-\$3]
-.        \}\}
-.        el \{\
-.          ds doc-volume "\*[doc-volume-operating-system]
-.          if \A\$3 \{\
-.            if d doc-volume-as-\$3 \
-.              as doc-volume "/\*[doc-volume-as-\$3]
-.          \}
-.          as doc-volume " \*[doc-volume-ds-\$2]
+.        ds doc-volume "\*[doc-volume-ds-\$2]
+.        if \A\$3 \{\
+.          if d doc-volume-as-\$3 \
+.            as doc-volume " (\*[doc-volume-as-\$3])
 .    \}\}\}
 .    el \{\
 .      ie "\$2"unass" \
@@ -379,7 +368,7 @@
 .        ds doc-volume DRAFT
 .      el .if "\$2"paper" \
 .        ds doc-volume UNTITLED
-.      \}\}
+.      \}
 .      if \A\$3 \{\
 .        if d doc-volume-ds-\$3 \
 .          ds doc-volume "\*[doc-volume-ds-\$3]
@@ -402,7 +391,7 @@
 .\" NS
 .\" NS override this in 'mdoc.local', if necessary
 .
-.ds doc-default-operating-system BSD
+.ds doc-default-operating-system OpenBSD
 .
 .
 .\" NS doc-operating-system global string
@@ -778,39 +767,14 @@
 .\" NS   doc-date-string
 .\" NS   doc-command-name
 .\" NS
-.\" NS local variables:
-.\" NS   doc-date-XXX
-.\" NS
 .\" NS width register 'Dd' set above
 .
-.ds doc-date-1  January
-.ds doc-date-2  February
-.ds doc-date-3  March
-.ds doc-date-4  April
-.ds doc-date-5  May
-.ds doc-date-6  June
-.ds doc-date-7  July
-.ds doc-date-8  August
-.ds doc-date-9  September
-.ds doc-date-10 October
-.ds doc-date-11 November
-.ds doc-date-12 December
-.
 .de Dd
 .  ds doc-command-name
-.
-.  ie \n[.$] \{\
-.    ie "\$1"$Mdocdate:" \
-.      ds doc-date-string \$2\~\$3, \$4
-.    el \{ .ie (\n[.$] == 3) \
-.      ds doc-date-string \$1\~\$2 \$3
-.    el \{\
-.      ds doc-date-string "\*[doc-date-\n[mo]]
-.      as doc-date-string \~\n[dy], \n[year]
-.    \}\}
-.  \}
+.  ie "\$1"$Mdocdate:" \
+.    ds doc-date-string \$2\~\$3, \$4
 .  el \
-.    ds doc-date-string Epoch
+.    ds doc-date-string \$*
 ..
 .
 .
@@ -976,7 +940,7 @@
 .\" NS width register 'Nd' set above
 .
 .de Nd
-.  nop \[em] \$*
+.  nop \[en] \$*
 ..
 .
 .
